### DISCUSSION TITLE
Allow diagnostic severity to be configured for rule groups

### DISCUSSION THREAD

**Discussion by tschaub** [biomejs/biome#2640] [Ideas]
I was surprised to see "correctness" diagnostics were emitted with "warning" severity and was hoping to find an easy way to configure all correctness rules to run at the "error" level instead.

I see that the CLI can be run with `--error-on-warnings` to treat all "warning" diagnostics as errors.  It could be nice to have more control over which groups were treated as errors.

For example, I was hoping I could do something like this:
```json
{
  "linter": {
    "enabled": true,
    "rules": {
      "correctness": {
        "all": "error"
      },
      "nursery": {
        "all": true
      }
    }
  }
}
```

Would it be possible for the `linter.rules.[group].all` to be configured with a `RulePlainConfiguration` in addition to a boolean?

**Comment by Conaclos** [comment]
> I was surprised to see "correctness" diagnostics were emitted with "warning" severity and was hoping to find an easy way to configure all correctness rules to run at the "error" level instead.

All recommended rules emit errors. Only [noUndeclaredVariables](https://biomejs.dev/linter/rules/no-undeclared-variables) and [useHookAtTopLevel](https://biomejs.dev/linter/rules/use-hook-at-top-level) are not recommended and in the correctness group. Are you referring to these rules?

We are planning to change this by using [error for correctness and warning for other groups](https://github.com/biomejs/biome/discussions/689#discussioncomment-7987718). We could also allow configuring the severity at a group level. It is still under discussion.

**Reply by tschaub** [reply]
Thanks for the detail, @Conaclos.  I didn't look through all the rules to see which were warning and which were error.  I noticed that both [undeclared](https://biomejs.dev/linter/rules/no-undeclared-variables/) and [unused](https://biomejs.dev/linter/rules/no-unused-variables/) variables were only generating warnings.

Does the documentation show the diagnostic level for these rules?  I may be missing it, but it wasn't obvious to me how to determine this.

**Reply by Conaclos** [reply]
On the documentation, the default severity of a rule corresponds to the severity of the diagnostics emitted in the examples.
